nflows/transforms/nonlinearities.py

```diff
@@ -447,3 +447,53 @@ def forward(self, inputs, context=None):
 
     def inverse(self, inputs, context=None):
         return self._spline(inputs, inverse=True)
+
+class HiLU(Transform):
+
+    def __init__(self, param_shape, init_log_range=0.0):
+        """
+        Hinge Linear Unit
+
+        similar to a PReLU, except that the slope is parameterized separately
+        for both positive and negative inputs/outputs
+        :param param_shape: Tuple of integers - shape of the parameter tensors including batch dim,
+                            must be broadcastable to input/output shape
+        :param init_range: Optional float: Randomize initial (log-scale) parameters by drawing them from a random
+                           uniform in the interval [-init_range, +init_range]
+        """
+        super().__init__()
+        self.param_shape = param_shape
+        self.init_log_range = init_log_range
+        self.pscale = nn.Parameter(torch.zeros(size=param_shape))
+        self.nscale = nn.Parameter(torch.zeros(size=param_shape))
+        if init_log_range>0.0:
+            with torch.no_grad():
+                self.pscale.data.copy_(torch.rand_like(self.pscale) * 2.0 * init_log_range - init_log_range)
```

Comment:
`torch.nn.init.uniform_` is a nicer way of doing this (and it takes care of doing `no_grad`).